packages/core/framework/src/http/utils/validate-query.ts

```diff
@@ -68,14 +68,19 @@ export function validateAndTransformQuery<TEntity extends BaseEntity>(
     next: NextFunction
   ) {
     try {
+      const restricted = req.restrictedFields.list()
```

Comment:
thought: Not sure if we do this already, but we should probably add a strong recommendation in our docs that you should use `validateAndTransformQuery` for custom endpoints in the store domain

Reply:
For best practice, it should probably be the recommendation for any custom endpoint. Same with `validateAndTransformBody`

Reply:
yes in theory we are already recommending that especially if we talk about the query config

Reply:
and then if people want something very custom they can just reuse the tools that we are using under the hood in those middleware. But then it is up to the user to manage everything if they don't use our recommended way

Reply:
Apparently, only the *body is mentioned in the documentation, should we ask shahed to have a look?

<img width="1139" alt="Screenshot 2024-10-22 at 11 41 13" src="https://github.com/user-attachments/assets/6b5641a8-c07d-44e0-94a5-0a5522aed45e">

Reply:
cc @shahednasser

Reply:
@olivermrbl do we also want to document ? or we do that later?
- `DEFAULT_STORE_RESTRICTED_FIELDS` from `framework/utils`
- `RestrictedFields` from `framework/http`
- medusa config `prjectConfig.http.restrictedFields`